src/controllers/hid/hidcontroller.cpp

```diff
@@ -137,6 +137,69 @@ int HidController::close() {
     return 0;
 }
 
+void HidController::processInputReport(int bytesRead) {
+    Trace process("HidController processInputReport");
+    unsigned char* pPreviousBuffer = m_pPollData[(m_PollingBufferIndex + 1) % kNumBuffers];
+    unsigned char* pCurrentBuffer = m_pPollData[m_PollingBufferIndex];
+    // Some controllers such as the Gemini GMX continuously send input reports even if it
+    // is identical to the previous send input report. If this loop processed all those redundant
+    // input report, it would be a big performance problem to run JS code for every  input report and
+    // would be unnecessary.
+    // This assumes that the redundant input report all use the same report ID. In practice we
+    // have not encountered any controllers that send redundant input report with different report
+    // IDs. If any such devices exist, this may be changed to use a separate buffer to store
+    // the last input report for each report ID.
+    if (bytesRead == m_LastPollSize &&
+            memcmp(pCurrentBuffer, pPreviousBuffer, bytesRead) == 0) {
+        return;
+    }
+    // Cycle between buffers so the memcmp below does not require deep copying to another buffer.
+    m_PollingBufferIndex = (m_PollingBufferIndex + 1) % kNumBuffers;
```

Comment:
The comment is now incorrect because the memcmp is *above*, not *below*. Does this matter? I am not sure. Let's move it back above the memcmp to be safe.

Reply:
I think, the comment is still correct. But it can be made more clear.